### ISSUE TITLE
Questions on Code Details: Image Size, Loss, Virtual Batch Multiplier

### ISSUE THREAD

**Issue by weimingwill** [tstandley/taskgrouping#6]
"REMEMBER to set your image size to 3x299x299 for both test and validation" in https://github.com/tstandley/taskgrouping/blob/dc6c89c269021597d222860406fa0fb81b02a231/model_definitions/xception_taskonomy_small.py#L12

Is it still applicable? I didn't find the hints to set such image size in source code or README.

**Comment by tstandley** [comment]
Oh boy. That comment needs to be removed. It's from the original xception
on imagenet. Unfortunately, I'm unable to login to GitHub at the moment.

**Comment by weimingwill** [comment]
Noted. Thank you for your clarification. I have two other questions regarding the code details and hope to seek your advice.

1. "Not taking the average of loss", Is it applicable only to `xception_taskonomy_new`? Or also applicable to other models such as `xception_taskonomy_small`? https://github.com/tstandley/taskgrouping/blob/dc6c89c269021597d222860406fa0fb81b02a231/taskonomy_losses.py#L163

2. How to set `virtual-batch-multipler? What's the value that is set in the paper? 1?
https://github.com/tstandley/taskgrouping/blob/dc6c89c269021597d222860406fa0fb81b02a231/train_taskonomy.py#L88

Thank you.

**Comment by tstandley** [comment]
1. The sum, not average, should be used whenever the TrevorRepFun is used
in the model to distribute backpropagation to tasks. Looking over the code,
this is only done for xception_taskonomy_new. For xception_taskonomy_small
the gradients are added from all the losses, so the average would probably
lead to more reproducible models. I don't remember what I did for that part
in the paper, but it only matters if you are trying to train your own.

2. The vba should be set to 1 unless you are trying to reproduce results
from a multi-gpu machine with a single gpu machine. For example, if you
train one model on a machine with 2 gpus, and another model on a machine
with 1 gpu, and you want to compare them, and you set the batch size to be
the maximum that will fit into ram, you should use -vba=2 for the single
gpu machine.

What it does is just run the forward-backward part multiple times before
doing an optimizer update.

**Comment by weimingwill** [comment]
Thank you.
